packages/data-broker/src/tests/services/weather/DateSanitiser.test.js

```diff
@@ -0,0 +1,123 @@
+import { expect } from 'chai';
+import { DateSanitiser } from '../../../services/weather/DateSanitiser';
+import { mockNow, resetMocks } from './testutil';
+
+describe('DateSanitiser', () => {
+  beforeEach(() => {
+    mockNow(1549360800 * 1000); // (2019-02-05 10:00 UTC)
+  });
+
+  afterEach(() => {
+    resetMocks();
+  });
+
+  it('throws when start > end', () => {
+    const functionCall = () => new DateSanitiser().sanitise('2019-01-20', '2019-01-10');
+    expect(functionCall).to.throw('Start date must be before (or equal to) end date');
+  });
+
+  describe('limits on earliest data', () => {
+    /*
+     * This is a limitation of the WeatherBit plan we use, if we request further back it falls over
+     */
+
+    it('both start and end dates are before the earliest available data', () => {
+      const { startDate: actualStartDate, endDate: actualEndDate } = new DateSanitiser().sanitise(
+        '2017-02-10',
+        '2017-02-20',
+      );
+
+      expect(actualStartDate).to.equal('2018-02-05'); // earliest date
+      expect(actualEndDate).to.equal('2018-02-06'); // earliest + 1 day
+    });
+
+    it('only start date is before the earliest available data', () => {
+      const { startDate: actualStartDate, endDate: actualEndDate } = new DateSanitiser().sanitise(
+        '2017-02-10',
+        '2019-01-01',
+      );
+
+      expect(actualStartDate).to.equal('2018-02-05'); // earliest date
+      expect(actualEndDate).to.equal('2019-01-01');
+    });
+
+    it('only start date is before earliest, end date is already earliest', () => {
+      const { startDate: actualStartDate, endDate: actualEndDate } = new DateSanitiser().sanitise(
+        '2017-02-10',
+        '2018-02-06',
```

Comment:
From the comment in the `it`, I thought this would be 
```suggestion
        '2018-02-05',
```
I.e., the end date is the earliest date

Reply:
Updated